platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java

```diff
@@ -62,7 +62,7 @@ public int getEndLocation()
         else
         {
 //            Rectangle bounds = attachedWidget.getBounds();
-            Rectangle bounds = new Rectangle();
+            Rectangle bounds;
```

Comment:
I would split the declaration and assignment into:
Widget sourceWidget;

And then refactor the code to remove the use of bounds..